### PR TITLE
PacketSource should contain `.to_hdf()` method

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

**Type:**  :rocket: `feature` 
Resolves https://github.com/tardis-sn/tardis/issues/2438 

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: tardis/montecarlo/tests/test_packet_source.py [hunk 1]
```diff
@@ -43,27 +52,41 @@ def blackbodysimplesource(self, request):
...
 
+    # def test_blackbody_regressions(
+    #     self, blackbodysimplesource
+    # ):
+    #     expected = pd.read_hdf(
+    #         self.regression_data.fpath, key="/black_body_simple_source/scalars"
+    #     )
+    #     np.testing.assert_allclose(
+    #         expected.base_seed, blackbodysimplesource.base_seed
+    #     )
```
Comment (atharva-2001) [inline tardis/montecarlo/tests/test_packet_source.py:88]: @andrewfullard Do we want to keep this test? Or do I need to change some other test module to test this change?

Comment (andrewfullard) [reply tardis/montecarlo/tests/test_packet_source.py:88]: If we could test this for the entire source output, that would be very useful.

Comment (atharva-2001) [reply tardis/montecarlo/tests/test_packet_source.py:88]: the packet source at this moment only has the base seed, other attributes are none I'm afraid

**Review by andrewfullard — DISMISSED** [review]
Great work getting tests in!

File: tardis/montecarlo/tests/test_packet_source.py [hunk 2]
```diff
@@ -44,98 +28,97 @@ def blackbodysimplesource(self, request):
...
+        assert_allclose(expected_bb.radius, actual_bb.radius)
+
+
+class TestBlackBodySimpleSourceRel:
     @pytest.fixture(scope="class")
-    def blackbody_simplesource_relativistic(self, request):
+    def blackbody_simplesource_relativistic(self):
         """
-        Create BlackBodySimpleSourceRelativistic instance.
+        Create BlackBodySimpleSource instance.
```
Comment (andrewfullard) [inline tardis/montecarlo/tests/test_packet_source.py:73]: This should still be relativistic, right?

File: tardis/montecarlo/tests/test_packet_source.py [hunk 3]
```diff
@@ -44,98 +28,97 @@ def blackbodysimplesource(self, request):
...
 
         Yields
         -------
-        tardis.montecarlo.packet_source.BlackBodySimpleSourceRelativistic
+        tardis.montecarlo.packet_source.BlackBodySimpleSource
```
Comment (andrewfullard) [inline tardis/montecarlo/tests/test_packet_source.py:77]: Here too

File: tardis/montecarlo/tests/test_packet_source.py (hunk 2 above)
Comment (atharva-2001) [reply tardis/montecarlo/tests/test_packet_source.py]: Thanks Andrew, I fixed this in https://github.com/tardis-sn/tardis/pull/2584/commits/9983c82267757fd1fe11f97eae9a16b0d5bde6f1

**Review by andrewfullard — CHANGES_REQUESTED** [review]
Looks like there's a merge conflict now

**Review by andrewfullard — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Review by jvshields — APPROVED** [review]

**Pull request merged**